src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +46,143 @@ public void editPerson() {
         editPersonDialog.enterNewValues(newAlice);
         editPersonDialog.pressEnter();
 
-        assertTrue(descOnFail(alicePersonCard, newAlice), alicePersonCard.isSamePerson(newAlice));
-        //TODO: write a more elegant function that uses assertEquals but compare two different types of objects
+        mainGui.sleep(1, TimeUnit.SECONDS);
+
+        //Confirm pending state correctness
+        assertTrue(alicePersonCard.isPendingStateRootVisible());
+        assertTrue(alicePersonCard.isPendingStateLabelVisible());
+        assertFalse(alicePersonCard.isPendingStateProgressIndicatorVisible());
+        assertTrue(alicePersonCard.getPendingStateLabel().equals("Editing"));
+
+        //Confirm the right card is selected after the edit
+        assertEquals(alicePersonCard, newAlice);
+        assertEquals(1, personListPanel.getSelectedCards().size());
+        assertEquals(alicePersonCard, personListPanel.getSelectedCards().get(0));
 
-        //TODO: Confirm the right card is selected after the edit, as follows
-        //assertEquals(1, personListPanel.getSelectedCards().size();
-        //assertEquals(alicePersonCard, personListPanel.getSelectedCards().get(0);
 
         //Confirm right values are displayed after grace period is over
-        guiRobot.sleep(5000); //wait for grace period
-        //TODO: link up to grace period
-        //TODO: confirm the person card display behave as expected during grace period
-        assertTrue(descOnFail(alicePersonCard, newAlice), alicePersonCard.isSamePerson(newAlice));
+        mainGui.sleepForGracePeriod();
+        assertEquals(alicePersonCard, newAlice);
 
         //Confirm the underlying person object has the right values
         assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
 
-        //TODO: confirm again after the next sync
-        //TODO: confirm other cards are unaffected
-        //TODO: confirm the right card is selected after the edit
-        //TODO: confirm status bar is updated correctly
+
+        //Confirm again after the next sync
+        mainGui.sleep(getTestingConfig().getUpdateInterval(), TimeUnit.MILLISECONDS);
+        assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
+
+        //Confirm other cards are unaffected.
+        personListPanel.clickOnListView();
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(2); //To make sure the card is visible
+        assertEquals(personListPanel.getPersonCardHandle(1), td.benson);
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(3);
+        assertEquals(personListPanel.getPersonCardHandle(2), td.charlie);
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(4);
+        assertEquals(personListPanel.getPersonCardHandle(3), td.dan);
```

Comment:
In that case, We couldn't make use of the assertEquals method? As this method give more details on the assert failure.